### PR TITLE
Adapt Rust's RFC process for Sway

### PR DESCRIPTION
**Pull request by sezna**
This is an adaption of Rust's RFC process for Sway. I've _slightly_ modified the template, readme, and initial RFC to describe the RFC process itself here.

### CODE REVIEW

File: rfcs/0001-rfc-process.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+followed consistently to introduce features to Sway.
+
+# Detailed design
+
+Many changes, including bug fixes and documentation improvements can be
+implemented and reviewed via the normal GitHub pull request workflow.
+
+Some changes though are "substantial", and we ask that these be put
+through a bit of a design process and produce a consensus among the Sway  
+community and the [core team].
```
Comment (mohammadfawaz) [inline rfcs/0001-rfc-process.md:33]:
```suggestion
through a bit of a design process and produce a consensus among the Sway
community and the [core team].
```
This was just causing some weird formatting issues in the preview.

File: rfcs/0001-rfc-process.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+merging the pull request, at which point the RFC is 'active', or
+reject it by closing the pull request.
+
+Whomever merges the RFC should do the following:
+
+* Assign an id, using the PR number of the RFC pull request. (If the RFC
+  has multiple pull requests associated with it, choose one PR number,
+  preferably the minimal one.)
+* Add the file in the `rfcs/` directory.
+* Create a corresponding issue on [Sway repo](https://github.com/FuelLabs/sway)
```
Comment (mohammadfawaz) [inline rfcs/0001-rfc-process.md:89]:
```suggestion
* Create a corresponding issue on [Sway repo](https://github.com/FuelLabs/sway).
```

File: rfcs/0001-rfc-process.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+* Fill in the remaining metadata in the RFC header, including links for
+  the original pull request(s) and the newly created Sway issue.
+* Add an entry in the [Active RFC List] of the root `README.md`.
```
Comment (mohammadfawaz) [inline rfcs/0001-rfc-process.md:92]: Might as well add this (empty) list already in the `README.md` as part of this PR?

File: rfcs/0001-rfc-process.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+submit an RFC first.
+
+## What the process is
+
+In short, to get a major feature added to Sway, one must first get the
+RFC merged into the RFC repo as a markdown file. At that point the RFC
+is 'active' and may be implemented with the goal of eventual inclusion
+into Sway.
+
+* Fork the RFC repo https://github.com/FuelLabs/sway-rfcs (or make a branch if you are core-team)
```
Comment (mohammadfawaz) [inline rfcs/0001-rfc-process.md:69]:
```suggestion
* Fork the RFC repo https://github.com/FuelLabs/sway-rfcs (or make a branch if you are core-team).
```

File: rfcs/0001-rfc-process.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+* Copy `0000-template.md` to `rfcs/0000-my-feature.md` (where
+'my-feature' is descriptive. don't assign an RFC number yet).
+* Fill in the RFC
```
Comment (mohammadfawaz) [inline rfcs/0001-rfc-process.md:72]:
```suggestion
* Fill in the RFC.
```

**Review by mohammadfawaz — COMMENTED** [review]
Looks good! Just some minor nits.

**Review by mitchmindtree — COMMENTED** [review]
Nice one @sezna, looking really good!

This will also be a great place to point folk who want to see certain features/changes in Sway as it provides some structure on how to propose them in a useful manner.

Would give the tick, but will defer to @mohammadfawaz after addressing his last suggestion.

File: rfcs/0001-rfc-process.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
+- Start Date: 2022-06-15
+- RFC PR: [FuelLabs/sway#1](https://github.com/FuelLabs/sway-rfcs/pull/1)
+
+_This RFC process is inspired by, and mostly copied from, Rust's RFC process. 
+See [here](https://github.com/rust-lang/rfcs/blob/master/text/0002-rfc-process.md)
+for Rust's version of this.
+
+Rust's RFC process has proven efficiency that we'd like to emulate._
```
Comment (adlerjohn) [inline rfcs/0001-rfc-process.md:8]: This doesn't italicize the block.

File: rfcs/0001-rfc-process.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+     stdlib design as well.
+
+Some changes do not require an RFC:
+
+   - Rephrasing, reorganizing, refactoring, or otherwise "changing shape
+does not change meaning".
+   - Additions that strictly improve objective, numerical quality
+criteria (warning removal, speedup, better platform coverage, more
+parallelism, trap more errors, etc.)
+   - Additions only likely to be _noticed by_ other developers-of-sway,
```
Comment (adlerjohn) [inline rfcs/0001-rfc-process.md:55]: I know the Rust version says `users-of-rust` lowercase, but that seems wrong.
```suggestion
   - Additions only likely to be _noticed by_ other developers-of-Sway,
```

**Review by mohammadfawaz — APPROVED** [review]
Looks good!

**Review by adlerjohn — APPROVED** [review]
utACK

**Comment by emilyaherbert** [comment]
I was OOO yesterday, can you hold off merging until I've had a chance to review?

**Review by emilyaherbert — CHANGES_REQUESTED** [review]
Maybe I missed it, but I didn't see anything specific about RFC's that propose breaking changes. It might be worthwhile to make a note requesting that any such RFC's describe the implications of the breaking change (i.e. if there is some syntax that changes, that change will need to be propagated to the standard library)

**Comment by sezna** [comment]
Does [this section](https://github.com/FuelLabs/sway-rfcs/pull/1/files#diff-9ae0eaba1ea4e62706bea031447886b41a1a9f14b132937f4fd801e9a8148042R29) cover what you're referring to? In Rust RFCs, this is typically where things like stdlib implementations are discussed [example](https://rust-lang.github.io/rfcs/3058-try-trait-v2.html#standard-implementations)

Additionally, breaking-ness is usually covered in the drawbacks or implementation details section. [example](https://rust-lang.github.io/rfcs/1212-line-endings.html#drawbacks). 

Are you proposing a specific section regarding addressing if anything is breaking, after all the other sections? I think that's definitely reasonable

**Pull request merged**